aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -1205,14 +1208,15 @@ std::tuple<Tensor, Tensor> _symeig_helper_cuda(const Tensor& self, bool eigenvec
   auto self_sizes = self.sizes().vec();
   self_sizes.pop_back();
 
-  // We create temporary tensors on the CPU, because tensors on the GPU
-  // cause segfault when passed to magmaSymeig. The data is later
-  // moved to the appropriate device.
+  // magmaSymeig uses a hybrid CPU-GPU algorithm to compute the eigenvalues and eigenvectors.
+  // The driver routine magma_(d/s)syev_gpu accepts a tensor on the CPU for eigvalenvalues.
+  // For better performance, we create a tensor with pinned_memory set to true.
+  // The data is later moved to the appropriate device.
   // In the case where self.numel() == 0, we just return an empty tensor of
   // dimensions on the CUDA (to avoid the unnecessary "to(at::kCUDA)")
   auto eigvals_working_copy = self.numel() == 0
                               ? at::empty(self_sizes, self.options())
-                              : at::empty(self_sizes, self.options().device(at::kCPU));
+                              : at::empty(self_sizes, at::TensorOptions(at::kCPU).dtype(self.dtype()).pinned_memory(true));
```

Comment:
I don't know enough about the tradeoffs of pinned memory to say if this is a good change or not. Do you have benchmarks on the performance of this?

A quick search through our codebase tells me that usually the user requests to use pinned memory.

Reply:
I didn't benchmark this yet. The rationale behind this change was primarily because the MAGMA documentation suggested this, and pinned memory was used in the older implementations in THC.

I will benchmark these ops with and without pinned memory, and post the results here.

Reply:
There isn't a conspicuous improvement in the benchmark timings - there actually is a slight regression is some cases. I'll revert to the older versions that don't use pinned memory.

More details about the benchmarks can be found [here](https://gist.github.com/vishwakftw/97a064cf13fb42d3c253f69415d83f5e).

Reply:
Interesting, thanks for running the numbers